compiler/rustc_mir_build/src/build/expr/into.rs

```diff
@@ -146,7 +146,7 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
                         this.then_else_break(
                             block,
                             lhs,
-                            Some(condition_scope),
+                            Some(condition_scope), // Temp scope
                             condition_scope,
                             source_info,
                             true,
```

Comment:
I didn't add a comment to this `true`, because I *think* that let expressions can't be encountered here, so the value shouldn't matter either way.

(But I wasn't confident enough to write that as a comment.)